sysdump/sysdump.go

```diff
@@ -0,0 +1,815 @@
+// Copyright 2021 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sysdump
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/cilium/cilium-cli/internal/utils"
+	"github.com/cilium/workerpool"
+	"github.com/hashicorp/go-multierror"
+	archiver "github.com/mholt/archiver/v3"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+)
+
+type SysdumpOptions struct {
```

Comment:
all the public structs (and their members :O) technically require comments to properly lint

Reply:
https://github.com/cilium/cilium-cli/pull/229 should catch this.

Reply:
Makes sense :+1:

Reply:
Done!

Reply:
Note that link would also complain about the following:
```
type name will be used as sysdump.SysdumpOptions by other packages, and that stutters; consider calling this Options
```